robot/client/client.go

```diff
@@ -333,6 +333,31 @@ func New(ctx context.Context, address string, clientLogger logging.ZapCompatible
 		}, rc.activeBackgroundWorkers.Done)
 	}
 
+	// If running in a testing environment, wait for machine to report a state of
+	// running. We often establish connections in tests and expected resources to
+	// be immediately available once the web service has started; resources will
+	// not be available when the machine is still initializing.
+	//
+	// It is expected that golang SDK users will handle lack of resource
+	// availability due to the machine being in an initializing state themselves.
+	if testing.Testing() {
```

Comment:
This is the "check" we discussed offline. See the comment above for an explanation; let me know if we're not on the same page about this logic not be required outside of a testing environment.